File: Proyecto/DataAccess/Repositories/EmployeeRepository.cs
using DataAccess.Contracts;
using DataAccess.Entities;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MySql.Data.MySqlClient;

namespace DataAccess.Repositories
{
    public class EmployeeRepository : MasterRepository, IEmployeeRepository
    {

        private string selectAll;
        private string insert;
        private string update;
        private string delete;


        public EmployeeRepository()
        {
            selectAll = "SELECT * FROM Employee";
            insert = "insert INTO Employee (IdNumber, Name, Mail, Birthday) VALUES (@idNumber, @name, @mail, @birthday);";
            update = "UPDATE Employee SET IdNumber = @idNumber, Name = @name, Mail =@mail, BirthDay = @birthday WHERE idPk = @idPK";
            delete = "DELETE FROM Employee WHERE idPk = @idPK";
        }

        public int Add(Employee entity)
        {

            parameters = new List<MySqlParameter>();
            parameters.Add(new MySqlParameter("@idNumber", entity.idNumber));
            parameters.Add(new MySqlParameter("@Name", entity.name));
            parameters.Add(new MySqlParameter("@mail", entity.mail));
            parameters.Add(new MySqlParameter("@birthday", entity.birthday));
            return ExecuteNonQuery(insert);
        }

        public int Edit(Employee entity)
        {
            parameters = new List<MySqlParameter>();
            parameters.Add(new MySqlParameter("@idPK", entity.idPK));
            parameters.Add(new MySqlParameter("@idNumber", entity.idNumber));
            parameters.Add(new MySqlParameter("@Name", entity.name));
            parameters.Add(new MySqlParameter("@mail", entity.mail));
            parameters.Add(new MySqlParameter("@birthday", entity.birthday));
            return ExecuteNonQuery(update);
        }

        public IEnumerable<Employee> GetAll()
        {
            var tableResult = ExecuteReader(selectAll);
            var listEmployees = new List<Employee>();
            foreach (DataRow item in tableResult.Rows)
            {
                listEmployees.Add(new Employee
                {
                    idPK = Convert.ToInt32(item[0]),
                    idNumber = item[1].ToString(),
                    name = item[2].ToString(),
                    mail = item[3].ToString(),
                    birthday = Convert.ToDateTime(item[4])
                });
            }

            return listEmployees;
        }

        public int Remove(int idPk)
        {
            parameters = new List<MySqlParameter>();
            parameters.Add(new MySqlParameter("@idPK", idPk));
            return ExecuteNonQuery(delete);
        }
    }
}

File: CrudLayered_VB/DataAccess/Repositories/EmployeeRepository.vb
Imports MySql.Data.MySqlClient
Imports DataAccess

Public Class EmployeeRepository
    Inherits MasterRepository
    Implements IEmployeeRepository


    Private selectAll As String
    Private insert As String
    Private update As String
    Private delete As String

    Public Sub New()
        selectAll = "SELECT * FROM Employee"
        insert = "insert INTO Employee (IdNumber, Name, Mail, Birthday) VALUES (@idNumber, @name, @mail, @birthday);"
        update = "UPDATE Employee SET IdNumber = @idNumber, Name = @name, Mail = @mail, Birthday = @birthday WHERE IdPK = @idPk"
        delete = "DELETE FROM Employee WHERE IdPk = @idPk"
    End Sub



    Public Function GetAll() As IEnumerable(Of Employee) Implements IGenericRepository(Of Employee).GetAll
        Dim resultTable = ExecuteReader(selectAll)
        Dim listEmployees = New List(Of Employee)

        For Each item As DataRow In resultTable.Rows
            listEmployees.Add(New Employee With {
                .IdPk = item(0),
                .IdNumber = item(1),
                .Name = item(2),
                .Mail = item(3),
                .Birthday = item(4)
             })
        Next
        Return listEmployees
    End Function

    Public Function Add(entity As Employee) As Integer Implements IGenericRepository(Of Employee).Add
        parameters = New List(Of MySqlParameter)
        parameters.Add(New MySqlParameter("@idNumber", entity.IdNumber))
        parameters.Add(New MySqlParameter("@name", entity.Name))
        parameters.Add(New MySqlParameter("@mail", entity.Mail))
        parameters.Add(New MySqlParameter("@birthday", entity.Birthday))
        Return ExecuteNonQuery(insert)
    End Function

    Public Function Edit(entity As Employee) As Integer Implements IGenericRepository(Of Employee).Edit
        parameters = New List(Of MySqlParameter)
        parameters.Add(New MySqlParameter("@idPk", entity.IdPk))
        parameters.Add(New MySqlParameter("@idNumber", entity.IdNumber))
        parameters.Add(New MySqlParameter("@name", entity.Name))
        parameters.Add(New MySqlParameter("@mail", entity.Mail))
        parameters.Add(New MySqlParameter("@birthday", entity.Birthday))
        Return ExecuteNonQuery(update)
    End Function

    Public Function Remove(id As Integer) As Integer Implements IGenericRepository(Of Employee).Remove
        parameters = New List(Of MySqlParameter)
        parameters.Add(New MySqlParameter("@idPk", id))

        Return ExecuteNonQuery(delete)
    End Function

    Public Function GetBySalary() As IEnumerable(Of Employee) Implements IEmployeeRepository.GetBySalary
        Throw New NotImplementedException()
    End Function
End Class
